Python/bytecodes.c

```diff
@@ -1339,29 +1366,39 @@ dummy_func(
             Py_DECREF(oldobj);
         }
 
-        inst(LOAD_CLASSDEREF, ( -- value)) {
-            PyObject *name, *locals = LOCALS();
-            assert(locals);
+        op(_LOAD_CLASSDEREF_INTRO, (-- class_dict)) {
+            class_dict = Py_NewRef(LOCALS());
```

Comment:
If `LOCALS()` can be NULL, then this segfaults.

Reply:
This is from the existing code for LOAD_CLASSDEREF. I can put in an assertion or SystemError if you like, though.

Reply:
This now uses LOAD_LOCALS, which throws a SystemError if there are no locals.